compiler/rustc_mir_build/src/check_unsafety.rs

```diff
@@ -124,7 +124,8 @@ impl<'tcx> UnsafetyVisitor<'_, 'tcx> {
     /// Handle closures/generators/inline-consts, which is unsafecked with their parent body.
     fn visit_inner_body(&mut self, def: LocalDefId) {
         if let Ok((inner_thir, expr)) = self.tcx.thir_body(def) {
-            let inner_thir = &inner_thir.borrow();
+            let _ = self.tcx.ensure_with_value().mir_built(def);
```

Comment:
```suggestion
            self.tcx.ensure_with_value().mir_built(def);
```
Could you add a comment explaining that the other dependent queries are recursively forced by mir_built ?